src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -311,20 +328,31 @@ export class PatientViewPageStore {
     });
 
     readonly mutationMolecularProfileId = remoteData({
-        await: () => [this.molecularProfilesInStudy],
+        await: () => [this.mutationMolecularProfile],
         invoke: async () => {
-            const profile = findMutationMolecularProfile(
-                this.molecularProfilesInStudy,
-                this.studyId
-            );
-            if (profile) {
-                return profile.molecularProfileId;
+            if (this.mutationMolecularProfile.result) {
```

Comment:
you can assume that result is populated because it's in await.   so, we only need this check if undefined is a valid